rfcs/text/0033-tty-output.md

```diff
@@ -0,0 +1,159 @@
+-->
+
+```
+  {
+    event: {
+      kind: 'event',
+      action: 'io'
+    },
+    process: {
+      args: ['ls'],
+      executable: '/bin/ls',
+      ...other_process_details,
+
+      entry_leader: <entry_context>,
+      session_leader: <session_context>,
+
+      // nested IO fieldset
+      io: {
+        fd: 1,
+        data: [
+          { time: 0, value: 'hello world' },
+          { time: 10, value: '#!/bin/bash' },
+          { time: 100, value: 'goodbye world' }
+        ]
+      }
+    }
+  }
+```
+
+<!--
+Stage 2: Included a real world example source document. Ideally this example comes from the source(s) identified in stage 1. If not, it should replace them. The goal here is to validate the utility of these field changes in the context of a real world example. Format with the source name as a ### header and the example document in a GitHub code block with json formatting, or if on the larger side, add them to the corresponding RFC folder.
+-->
+
+<!--
+Stage 3: Add more real world example source documents so we have at least 2 total, but ideally 3. Format as described in stage 2.
+-->
+
+## Scope of impact
+
+<!--
+Stage 2: Identifies scope of impact of changes. Are breaking changes required? Should deprecation strategies be adopted? Will significant refactoring be involved? Break the impact down into:
+ * Ingestion mechanisms (e.g. beats/logstash)
+ * Usage mechanisms (e.g. Kibana applications, detections)
+ * ECS project (e.g. docs, tooling)
+The goal here is to research and understand the impact of these changes on users in the community and development teams across Elastic. 2-5 sentences each.
+-->
+
+## Concerns
+
+<!--
+Stage 1: Identify potential concerns, implementation challenges, or complexity. Spend some time on this. Play devil's advocate. Try to identify the sort of non-obvious challenges that tend to surface later. The goal here is to surface risks early, allow everyone the time to work through them, and ultimately document resolution for posterity's sake.
+-->
+
+1. Data exfiltration. TTY output is a sensitive surface area to expose. It definitely should be featured gated, and an opt in for customers. At least until we implement some robust "regex scrubbing" mechanisms.
+2. Per event batch size should be considered. If batch size is too big, alerting on IO data becomes fuzzy, as rules are evaluated on the document level, not on an indiviual line of output. This could make it a challenge to figure out what part of the message triggered the alert. Also, some performance benchmarking should be done to see how rule evaluation performs on process.io.data.value when data.length > X
```

Comment:
```suggestion
2. Per event batch size should be considered. If batch size is too big, alerting on IO data becomes fuzzy, as rules are evaluated on the document level, not on an individual line of output. This could make it a challenge to figure out what part of the message triggered the alert. Also, some performance benchmarking should be done to see how rule evaluation performs on process.io.data.value when data.length > X
```
typo